internal/controller/supervisorconfig/githubupstreamwatcher/github_upstream_watcher_test.go

```diff
@@ -61,6 +61,32 @@ var (
 	githubIDPKind = idpv1alpha1.SchemeGroupVersion.WithKind("GitHubIdentityProvider")
 )
 
+type fakeGithubDialer struct {
+	t            *testing.T
+	realAddress  string
+	realCertPool *x509.CertPool
+}
+
+func (f fakeGithubDialer) IsReachableAndTLSValidationSucceeds(ctx context.Context, address string, _ *x509.CertPool, logger plog.Logger) error {
+	require.Equal(f.t, "api.github.com:443", address)
+
+	return ptls.NewDialer().IsReachableAndTLSValidationSucceeds(ctx, f.realAddress, f.realCertPool, logger)
```

Comment:
It looks like it is calling the real production function. Is that intended?

Reply:
It is intended, when using this `fakeGithubDialer` the `realAddress` and `realCertPool` have to be provided, presumably as something other than `github.com`.